src/components/All-issues/AllIssues.tsx

```diff
@@ -0,0 +1,238 @@
+"use-client"
+import style from "./index.module.css"
+import style2 from "../main-content/general.module.css"
+import { useTheme } from "../contextApi/ThemeContext"
+import Image from "next/image"
+import * as Dialog from "@radix-ui/react-dialog"
+import { DiscIcon } from "@radix-ui/react-icons"
+import { useState } from "react"
+
+interface Labels2Element {
+    language: string
+    number: number
+    chat: {
+        chat_number: number
+        chat_icons: string
+    }
+    last_updated: string
+    icon: string
+}
+
+interface Data {
+    id: number
+    header: string
+    company: { smallIcon: string; name: string }
+    behaviour_text: string
+    expected_behaviour_text: string
+    labels: Array<string>
+    labels2: Labels2Element[]
+}
+interface AllIssuesProps {
+    setSearchParams: React.Dispatch<React.SetStateAction<any>>
+    issueList: Data[]
+    issues: Data[]
+    visibleIssues: Array<any>
+    setIssueList: React.Dispatch<React.SetStateAction<any>>
+    setCurrentPage: React.Dispatch<React.SetStateAction<any>>
+    searchParams: {
+        language: string
+        organisation: string
+        type: string
+        recent: string
+    }
+}
+
+interface Label2 {
+    language: string
+    number: number
+    chat: {
+        chat_number: number
+        chat_icons: string
+    }
+    last_updated: string
+    icon: string
+}
+
+interface VisibleIssues {
+    id: number
+    header: string
+    company: { smallIcon: string; name: string }
+    behaviour_text: string
+    expected_behaviour_text: string
+    labels: string[]
+    labels2: {
+        language: string
+        number: number
+        chat: {
+            chat_number: number
+            chat_icons: string
+        }
+        last_updated: string
+        icon: string
+    }[]
+}
+
+export default function AllIssues(props: AllIssuesProps) {
+    const { isDarkMode } = useTheme()
+    const [expandText, setExpandText] = useState<boolean[]>([])
```

Comment:
why do we have a boolean array? and what do each boolean in the array represent?

Reply:
This is to show the modal carrying the full description of issues at a specific index.

Reply:
The boolean state doesn't show what type of booleans in the array is to be passed. We should create a object key mapping for the boolean for better readability